aviary/subsystems/propulsion/propeller_performance.py

```diff
@@ -2,8 +2,135 @@
 import numpy as np
 
 from aviary.utils.aviary_values import AviaryValues
+from aviary.variable_info.enums import OutMachType
 from aviary.variable_info.variables import Aircraft, Dynamic
 from aviary.subsystems.propulsion.hamilton_standard import HamiltonStandard, PostHamiltonStandard, PreHamiltonStandard
+from aviary.subsystems.propulsion.propeller_map import PropellerMap
+
+
+class OutMachs(om.ExplicitComponent):
+    """This utility sets up relations among helical Mach, free stream Mach and propeller tip Mach.
+    helical_mach = sqrt(mach^2 + tip_mach^2).
+    It computes the value of one from the inputs of the other two.
+    """
+
+    def initialize(self):
+        self.options.declare("num_nodes", types=int)
+        self.options.declare(
+            "output_mach_type",
+            default=OutMachType.HELICAL_MACH,
+            types=OutMachType,
+            desc="get one type of Mach number from the other two",
+        )
+
+    def setup(self):
+        nn = self.options["num_nodes"]
+        out_type = self.options["output_mach_type"]
+        arange = np.arange(self.options["num_nodes"])
+
+        if out_type is OutMachType.HELICAL_MACH:
+            self.add_input(
+                "mach",
+                val=np.zeros(nn),
+                units="unitless",
+                desc="Mach number",
+            )
+            self.add_input(
+                "tip_mach",
+                val=np.zeros(nn),
+                units="unitless",
+                desc="tip Mach number at 3/4 of radius of a blade",
```

Comment:
Wait why is propeller tip speed defined as at 3/4 blade radius location? That seems like it will break a bunch of important assumptions - advance ratio uses tipspeed at actual blade tip, and constraining tip speed to avoid material or Mach constraints makes less sense if you are constraining 3/4 value (and don't know the actual maximum speed, which is of interest)

Reply:
I removed the phrase about 3/4. We will always use the tip.